### PR TITLE
[RFR] Rotate remaining pathcard [follows PR#11]

### PR DESCRIPTION
**Pull request by pemoreau**
Rotate remaining pathcard using Redux and optimistic rendering

screen shot 2018-10-24 at 15 40 05

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -62,17 +67,20 @@ build-front:
...
 
 build-server:
-	rm -rf $(DIST_SERVER)
-	mkdir -p $(DIST_SERVER)
-	cp $(FILES_SERVER) $(DIST_SERVER)
+	rm -rf $(DIST_SERVER) $(DIST_SRC)
+	mkdir -p $(DIST_SERVER) $(DIST_SRC)
+	./node_modules/.bin/babel -d dist $(FILES_SERVER)
+	./node_modules/.bin/babel -d dist $(FILES_SRC)
+	#cp $(FILES_SERVER) $(DIST_SERVER)
```
Comment (Kmaschta) [inline Makefile]: Commented code found

File: Makefile [hunk 2]
```diff
@@ -62,17 +67,20 @@ build-front:
...
 
 deploy-front: build-front
 	aws s3 rm s3://labyrinth-react --recursive
 	aws s3 sync $(DIST_CLIENT) s3://labyrinth-react/
 
 deploy-server: build-server
-	scp -r Makefile package.json docker-compose.yaml $(DIST_SERVER) labyrinth-react:app/
-	ssh labyrinth-react 'cd app && make stop-prod start-prod'
+	#ssh labyrinth-react 'rm -rf app/*' || true
```
Comment (Kmaschta) [inline Makefile]: Commented code found

File: package.json [hunk 3]
```diff
@@ -55,12 +56,18 @@
...
         "immer": "^1.7.2",
         "minimist": "^1.2.0",
+        "mongoose": "^5.3.4",
         "react": "^16.5.2",
         "react-dom": "^16.5.2",
-        "react-scripts": "2.0.5"
+        "react-redux": "^5.0.7",
+        "react-scripts": "2.0.5",
+        "redux": "^4.0.1",
+        "redux-thunk": "^2.3.0"
```
Comment (Kmaschta) [inline package.json]: Do you really need this lib ?

File: server/controllers/gameController.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        .catch(err => {
+            res.status(400).send('unable to save to database');
+        });
+};
+
+export const getGame = (req, res) => {
+    GameModel.findById(req.params.id, function(err, gameDocument) {
+        if (err) {
+            res.status(400).send('unable to get the game');
+        } else {
```
Comment (Kmaschta) [inline server/controllers/gameController.js]: Prefer to use early return instead of else.

File: src/config.js [hunk 5]
```diff
@@ -0,0 +1,2 @@
+export const apiUrl = 'http://localhost:3000';
+// export const apiUrl = 'http://ec2-34-244-162-222.eu-west-1.compute.amazonaws.com:3000';
```
Comment (Kmaschta) [inline src/config.js]: Commented code found.

Protip: Use environment variables for this kind of config. It's easy to do so thanks to docker compose.

File: package.json (hunk 3 above)
Comment (pemoreau) [reply package.json]: will try to remove this lib in next PR

File: src/config.js (hunk 5 above)
Comment (pemoreau) [reply src/config.js]: Hum... not so easy since client is not in docker on AWS S3

File: src/reducers/gameReducer.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+    ROTATE_REMAINING_PATHCARD_CLOCKWISE,
+    ROTATE_REMAINING_PATHCARD_CLOCKWISE_OPTIMISTIC,
+} from '../actions/types';
+
+export default function gameReducer(state = [], action) {
+    switch (action.type) {
+        case CREATE_GAME_SUCCESS:
+        case ROTATE_REMAINING_PATHCARD_CLOCKWISE:
+        case ROTATE_REMAINING_PATHCARD_CLOCKWISE_OPTIMISTIC:
+            return action.payload;
```
Comment (JulienMattiussi) [inline src/reducers/gameReducer.js]: If all types do same action, switch is not necessary
(And also type, almost)

File: src/components/PlayerCards.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+                        degrees={90 * remainingPathCard.direction}
+                        type={remainingPathCard.type}
+                        target={remainingPathCard.target}
+                        onClick={() => onRotateRemainingPathCard(game)}
+                    />
+                </div>
+                <div className="box" />
+                <div className="box" />
+                <div className="box" />
+            </div>
```
Comment (JulienMattiussi) [inline src/components/PlayerCards.js]: What are these empty box ?

File: src/components/App.css [hunk 8]
```diff
@@ -17,6 +17,12 @@ body {
     margin: 0 auto;
 }
 
+#start-new-game {
+    width: 100%;
+    height:400px;
```
Comment (JulienMattiussi) [inline src/components/App.css]: lint!

File: src/common/game.js [hunk 9]
```diff
@@ -184,10 +184,11 @@ const moveRemainingPathCardClockwise = game => moveRemainingPathCard(game, Direc
 
 const moveRemainingPathCardAntiClockwise = game => moveRemainingPathCard(game, Direction.WEST);
 
-const rotateRemainingPathCard = game => {
+// deltaDirection=1 for clokwise, -1 for anti-clockwise
```
Comment (JulienMattiussi) [inline src/common/game.js]: prefer using a constant instead of comment it

**Comment by JulienMattiussi** [comment]
"react-scripts": "2.0.5" is still in fixed

File: src/reducers/gameReducer.js (hunk 6 above)
Comment (Kmaschta) [reply src/reducers/gameReducer.js]: A reducer that ALL the actions of the app, so, it's better to have the switch IMHO.

Comment (JulienMattiussi) [reply src/reducers/gameReducer.js]: right
so keep it. 
My fault

**Pull request merged**